src/Android/MainActivity.cs

```diff
@@ -159,7 +142,16 @@ protected override void OnNewIntent(Intent intent)
             base.OnNewIntent(intent);
             try
             {
-                if (intent.GetBooleanExtra("generatorTile", false))
+                var uri = intent?.GetStringExtra("uri");
+                if (uri != null)
```

Comment:
I think we could make it one line (not sure if there is an standard that is against this):
```c#
if(intent?.GetStringExtra("uri") is string uri)
{
    ....
}
```